### DISCUSSION TITLE
The openapi schema fails to find path parameters if they are not defined as a function parameter

### DISCUSSION THREAD

**Discussion by OnyxMsi** [fastapi/fastapi#9987] [Show and tell]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
from fastapi.testclient import TestClient

app = FastAPI()


@app.get("/{first:int}/{second}")
def route(request: Request):
    # This works
    return request.path_params["second"]

client = TestClient(app)
response = client.get("/openapi.json")
data = response.json()
assert "parameters" in data["paths"]["/{first}/{second}"]["get"]
```

### Description

**This error concerns only the OpenAPI schema generation, the web part works fine.**

It happen when using https://www.starlette.io/routing/#path-parameters `request.path_params` in order to access path attributes instead of declaring them as function attribute.
When parsing the routes to generate the OpenAPI stuff, fastapi is missing
 them.

I’ve created a patch that fix this by editing the `get_dependant` function in `fastapi.dependencies.utils`.
[0001-OpenAPI-generation-supports-path-parameter-not-defin.txt](https://github.com/tiangolo/fastapi/files/12242495/0001-OpenAPI-generation-supports-path-parameter-not-defin.txt)

### Operating System

Linux

### Operating System Details

Linux **** 6.1.0-9-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.27-1 (2023-05-08) x86_64 GNU/Linux

### FastAPI Version

0.100.1

### Python Version

Python 3.11.2

**Comment by pete001** [comment]
Yep, we are seeing the same issue.

**Reply by pete001** [reply]
Can confirm this patch works perfectly 👌
